docs/UseCases.rst

```diff
@@ -0,0 +1,36 @@
+=====================
+Use Cases & Philosopy
+=====================
+
+.. warning:: Incorrect Information below
+
+    Until this document is reviewed and corrected, it is known to contain errors, especially w.r.t. philosopy.
+
+Philosopy
+=========
+
+Frost is intended to build upon the pytest framework to provide extra features to support "infrastructure auditing".
+
+The added features include:
+
+    - providing a standard way to inject meta data in the JSON test result file.
+    - providing a CLI wrapper (the ``frost`` program) to simplify execution of tests within the full supported frost environment.
+
+Note that the added features are all extensions to the normal pytest operations. Therefore, all tests can be executed directly from pytest. This can be highly useful for adhoc queries, and the development of new tests.
+
+Target Domain
+-------------
+
+Frost is intended to simplify the process of auditing existing system configurations, when the configuration data is accessible via an API. Frost assumes that the test results (in JSON format) will be acted upon by a subsequent non-Frost process.
+
+Since, in general, the number of configuration items to check is not statically known when the test is writen or executed, some lesser used features of pytest are needed. In particular, the test collection phase often requires API calls to the service to discover the complete tree (or network) of configuration items to be checked. While uncommon in internet examples, dynamic test discovery is fully supported by pytest.
+
+In practice, many of the same calls needed to determine the list of items may need to be repeated to collect the additional information needed by one or more tests. This leads to a tension between collecting all potentially needed data in the first (discovery) pass, or making many duplicate calls to the service as each test collects the additional information it requires.
```

Comment:
```suggestion
In practice, many of the same calls needed to determine the list of resources may need to be repeated to collect the additional information needed by one or more tests. This leads to a tension between collecting all potentially needed data in the first (discovery) pass, or making many duplicate calls to the service as each test collects the additional information it requires. Currently, Frost collects all need resources during the first pass and caches them for use by other tests within the same run of Frost and by subsequent runs of Frost. 
```

Reply:
I'll admit this a part that is still fuzzy for me. I think it may more accurately be expressed as "Frost best practices call for caching..." (or some such).

GitHub branches does the final "get all the details" at test time, not collection time. If I didn't, it'd be back to the massive "collect the world" approach that the current 10hr metrics runs take.

I'll work on this piece some more.

Reply:
@ajvb please see [this added material](https://github.com/hwine/frost/compare/hwine%2Fusecases%5E...hwine%2Fusecases?diff=unified#diff-5fe763154d28cda87f636430568dc6849423e32e791ec62e08d75bbd1171b81aR33-R40). 

Let me know what still needs to change.